### DISCUSSION TITLE
Is `fs` basically like a `tempdir` scoped to each test function?

### DISCUSSION THREAD

**Discussion by jamesbraza** [jmcgeheeiv/pyfakefs#815] [Q&A] [answered]
The following code runs:

```python
import pytest

def test_a(fs):
    with open("hi.txt", mode="w"):
        pass

def test_b(fs):
    with pytest.raises(FileNotFoundError, match="hi.txt"):
        with open("hi.txt"):
            pass
```

So just to confirm, can one think of each `fs` fixture as being a temporary OS (similar in concept to a temporary directory in `/tmp`)?

Why I am asking is because if the answer is yes, I think I don't need `with tempfile.TemporaryDirectory()` in my test files any more.

**Comment by mrbean-bremen** [comment, marked as answer]
Yes, this is exactly what it is. There are module and session-scoped variants (`fs_module` and `fs_session`), but the standard `fs` fixture basically creates a new empty (except for the temp dir) in-memory filesystem at each test start.

**Reply by jamesbraza** [reply]
Okay, this clears me up!  Thank you for the speedy response 👌
